### DISCUSSION TITLE
Support for Solana wallets like phantom

### DISCUSSION THREAD

**Discussion by anoushk1234** [rainbow-me/rainbowkit#344] [Ideas]
We could easily plug in solana wallet adapter and support for wallets like phantom and solflare which would truly make this a multichain solution

**Comment by nickbytes** [comment]
I like this idea. 

Based on my limited knowledge of how Solana works + current ecosystem, I think there's a bit of work to get support for Solana into RainbowKit. First, RainbowKit relies on [wagmi](https://wagmi.sh/) and [ethers](https://docs.ethers.io/v5/). Ethers does not support Solana, and to my knowledge, there does not exist a fork of Ethers for Solana. This is a major hurdle.

Once we had that Solana Ethers fork (or similar), we'd need to separate Ethereum-specific logic from wagmi core and react libraries.  This seems doable, especially given recent updates to modularize the wagmi codebase.

After that, I imagine there are still some challenges within the RainbowKit UI and core logic, plus some differences between Solana and EVM-chains, dapps, etc etc etc.

All that being said, I don't think this is impossible, and would frankly be excited to see community involvement and support to make this happen. If you and/or other Solana contributors want to put together an RFC laying out how you see this working, that'd be super cool. We might find out that the best path forward is a fork of the mentioned libraries (including RainbowKit), and that's okay too.

Pumped about pushing all developer tools (and user experience) forward, regardless of chain or wallet.

**Reply by anoushk1234** [reply]
Thanks for the response @nickbytes , I think we could get this to work without the ethers fork, but we would have to use Solana-wallet-adapter to do that and we can refer their [code](https://github.com/solana-labs/wallet-adapter/blob/master/packages/ui/react-ui/src/WalletModal.tsx) as they have also built a modal so we just have to integrate those functions into the rainbow modal. I could be wrong though since I haven't explored much of how Rainbow Kit is designed but ill try to implement it myself.

**Reply by nickbytes** [reply]
Cool, good luck! Excited to see what ya come up with here.

We'll hopefully be sharing more of what's on our roadmap over the coming weeks. We don't have Solana support on the roadmap currently, but I'm interested to see what it might take to make non-EVM chains work here.

**Comment by anoushk1234** [comment]
Hey @nickbytes we've implemented our own version of a unified wallet modal in our dapp https://app.metapasshq.xyz Try connecting your wallet, it supports Metamask, Wallet Connect and Solana Wallets and we use wagmi under the hood
The code is open source at https://github.com/Metapass/metapass-v2/tree/prod/components/Navigation
